go/vt/vttablet/tabletmanager/vreplication/vdbclient.go

```diff
@@ -83,17 +87,48 @@ func (vc *vdbClient) ExecuteFetch(query string, maxrows int) (*sqltypes.Result,
 	return vc.DBClient.ExecuteFetch(query, maxrows)
 }
 
-func (vc *vdbClient) Retry() error {
+// Execute is ExecuteFetch without the maxrows.
+func (vc *vdbClient) Execute(query string) (*sqltypes.Result, error) {
+	return vc.ExecuteFetch(query, 100000)
+}
+
+func (vc *vdbClient) ExecuteWithRetry(ctx context.Context, query string) (*sqltypes.Result, error) {
+	qr, err := vc.Execute(query)
+	for err != nil {
+		if sqlErr, ok := err.(*mysql.SQLError); ok && sqlErr.Number() == mysql.ERLockDeadlock || sqlErr.Number() == mysql.ERLockWaitTimeout {
+			log.Infof("retryable error: %v, waiting for %v and retrying", sqlErr, dbLockRetryDelay)
+			if err := vc.Rollback(); err != nil {
+				return nil, err
+			}
+			time.Sleep(dbLockRetryDelay)
+			// Check context here. Otherwise this can become an infinite loop.
+			select {
+			case <-ctx.Done():
+				return nil, io.EOF
+			default:
+			}
+			qr, err = vc.Retry()
+			continue
+		}
+		return qr, err
+	}
+	return qr, nil
+}
+
+func (vc *vdbClient) Retry() (*sqltypes.Result, error) {
+	var qr *sqltypes.Result
 	for _, q := range vc.queries {
 		if q == "begin" {
 			if err := vc.Begin(); err != nil {
-				return err
+				return nil, err
 			}
 			continue
 		}
-		if _, err := vc.DBClient.ExecuteFetch(q, 10000); err != nil {
-			return err
+		result, err := vc.DBClient.ExecuteFetch(q, 100000)
```

Comment:
Since this is using the default maxRows, can you replace it with a call to `Execute`?

Reply:
This was set to 100K rows based on the max packet size of 30K. There's actually no need to set a limit on number of rows because it can't exceed the max packet size. However, since max packet size is configurable, I think I should set it to max packet size itself (assuming worst case of 1 byte per row).

Reply:
Ah, this is being passed down to DBClient. That sounds like a good idea.

Reply:
As for maxrows in the original `ExecuteFetch`, the signature cannot be changed until old binlog player is deprecated.